src/workerd/api/pyodide/pyodide.c++

```diff
@@ -317,40 +317,60 @@ kj::Array<kj::String> ArtifactBundler::parsePythonScriptImports(kj::Array<kj::St
   return result.releaseAsArray();
 }
 
-// This is equivalent to `pkgImport.replace('.', '/') + ".py"`.
-kj::String importToModuleFilename(kj::StringPtr pkgImport) {
-  auto result = kj::heapString(pkgImport.size() + 3);
-  for (auto i = 0; i < pkgImport.size(); i++) {
-    if (pkgImport[i] == '.') {
-      result[i] = '/';
-    } else {
-      result[i] = pkgImport[i];
-    }
-  }
+const kj::Array<kj::StringPtr> snapshotImports = kj::arr("_pyodide"_kj,
+    "_pyodide.docstring"_kj,
+    "_pyodide._core_docs"_kj,
+    "traceback"_kj,
+    "collections.abc"_kj,
+    // Asyncio is the really slow one here. In native Python on my machine, `import asyncio` takes ~50
+    // ms.
+    "asyncio"_kj,
+    "inspect"_kj,
+    "tarfile"_kj,
+    "importlib",
+    "importlib.metadata"_kj,
+    "re"_kj,
+    "shutil"_kj,
+    "sysconfig"_kj,
+    "importlib.machinery"_kj,
+    "pathlib"_kj,
+    "site"_kj,
+    "tempfile"_kj,
+    "typing"_kj,
+    "zipfile"_kj);
+
+kj::Array<kj::StringPtr> ArtifactBundler::getSnapshotImports() {
+  return kj::heapArray(snapshotImports.begin(), snapshotImports.size());
+}
 
-  result[pkgImport.size()] = '.';
-  result[pkgImport.size() + 1] = 'p';
-  result[pkgImport.size() + 2] = 'y';
-  return result;
+bool isWorkerModuleImport(
+    kj::HashSet<kj::String>& workerModules, kj::ArrayPtr<const char> firstComponent) {
+  // TODO: handling for:
+  // 1. namespace packages
+  // 2. shared libraries
+  return workerModules.contains(kj::str(firstComponent, ".py")) ||
+      workerModules.contains(kj::str(firstComponent, "/__init__.py"));
```

Comment:
Aren't `__init__.py` files no longer required with new python versions to declare a module?

Reply:
You may be thinking of "namespace packages":
https://peps.python.org/pep-0420/#specification

Reply:
The relevant detail is that a namespace packages are never preferred over normal packages. So if there is a folder called fastapi with no `__init__.py` but also a package called fastapi, then the package will be preferred. If there is no package called fastapi, then the folder gets turned into a namespace package.

So ideally to handle this we need to tell this algorithm the list of imports defined by the installed packages. Since it's a bit more complicated, I figured it was okay to leave as a TODO for now.